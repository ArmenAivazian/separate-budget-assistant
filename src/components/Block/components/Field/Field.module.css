.field-name {
    font-weight: bold;
    text-align: right;
}