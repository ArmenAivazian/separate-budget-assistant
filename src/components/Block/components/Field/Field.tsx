import { ConfigValue } from "../../../../types/ConfigValue";
import style from "./Field.module.css";

export function Field({
  value,
  text,
  localStorageName,
  options,
  type = "number",
  setValue,
}: ConfigValue) {
  const handleChange = (val: string) => {
    setValue(val);

    if (localStorageName) localStorage.setItem(localStorageName, val);
  };

  return (
    <>
      <p className={style["field-name"]}>{text} :</p>
      {options ? (
        <select>
          {options.map((option) => (
            <option value={option}>{option}</option>
          ))}
        </select>
      ) : (
        <input
          value={value}
          onChange={({ target }) => handleChange(target.value)}
          type={type}
        />
      )}
    </>
  );
}
