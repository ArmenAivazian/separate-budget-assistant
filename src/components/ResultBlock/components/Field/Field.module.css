.name {
    text-align: right;
}

.value {
    text-align: left;
}

.topLine {
    border-top: 1px solid #000;
    padding-top: 5px;
    margin-top: 5px;
}