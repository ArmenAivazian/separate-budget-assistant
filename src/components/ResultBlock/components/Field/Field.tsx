import { FieldProps } from "./Field.types";
import style from "./Field.module.css";

export function Field({ withLine, text, value1, value2 }: FieldProps) {
  const nameStyle = withLine ? `${style.name} ${style.topLine}` : style.name;
  const valueStyle = withLine ? `${style.value} ${style.topLine}` : style.value;

  return value1 ? (
    <>
      <p className={nameStyle}>{text}</p>
      <p className={valueStyle}>[A] {value1}</p>
      <p className={valueStyle}>[N] {value2}</p>
      <p className={valueStyle}>[SUM] {value1 + (value2 || 0)}</p>
    </>
  ) : (
    <></>
  );
}
