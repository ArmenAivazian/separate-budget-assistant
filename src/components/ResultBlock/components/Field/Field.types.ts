export interface FieldProps {
  withLine?: boolean;
  text: string;
  value1?: number;
  value2?: number;
}
