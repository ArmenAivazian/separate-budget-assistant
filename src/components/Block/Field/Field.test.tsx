import renderer from "react-test-renderer";
import { Field } from "./Field";

const defaultProps = { value: "100", text: "text", setValue: () => {} };

describe("Block -> Field", () => {
  describe("snapshots", () => {
    it("default", () => {
      const component = renderer.create(<Field {...defaultProps} />);
      expect(component).toMatchSnapshot();
    });
    
    it("select", () => {
      const component = renderer.create(
        <Field options={["option1", "option2"]} {...defaultProps} />
      );
      expect(component).toMatchSnapshot();
    });
    
    it("date", () => {
      const component = renderer.create(
        <Field type="date" {...defaultProps} />
      );
      expect(component).toMatchSnapshot();
    });
  });
});
