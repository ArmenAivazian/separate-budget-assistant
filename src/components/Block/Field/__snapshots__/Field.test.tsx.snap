// Vitest Snapshot v1

exports[`Block -> Field > snapshots > date 1`] = `
[
  <p
    className="_field-name_10d9f7"
  >
    text
     :
  </p>,
  <input
    onChange={[Function]}
    type="date"
    value="100"
  />,
]
`;

exports[`Block -> Field > snapshots > default 1`] = `
[
  <p
    className="_field-name_10d9f7"
  >
    text
     :
  </p>,
  <input
    onChange={[Function]}
    type="number"
    value="100"
  />,
]
`;

exports[`Block -> Field > snapshots > select 1`] = `
[
  <p
    className="_field-name_10d9f7"
  >
    text
     :
  </p>,
  <select>
    <option
      value="option1"
    >
      option1
    </option>
    <option
      value="option2"
    >
      option2
    </option>
  </select>,
]
`;
